src/compute-client/src/service.rs

```diff
@@ -311,7 +325,32 @@ where
                             (PeekResponse::Error(e), _) => PeekResponse::Error(e),
                             (PeekResponse::Rows(mut rows), PeekResponse::Rows(r)) => {
                                 rows.extend(r.into_iter());
-                                PeekResponse::Rows(rows)
+
+                                let total_size = rows
+                                    .iter()
+                                    // Note: if the type of count changes in the future to be a
+                                    // signed integer, then we'll need to consolidate rows before
+                                    // taking this summation.
+                                    .map(|(row, count): &(Row, NonZeroUsize)| {
+                                        let size = row
+                                            .byte_len()
+                                            .saturating_add(std::mem::size_of_val(count));
+                                        u64::cast_from(size)
+                                    })
+                                    .fold(0u64, |total, x| total.saturating_add(x));
+                                match self.max_result_size {
+                                    Some(max_size) if total_size > max_size => {
```

Comment:
Probably want to move the computation of `total_size` into the match? I'm not sure actually because we always set the max_result_size, so we wouldn't save any CPU cycles... I don't feel strongly about this.

Reply:
I tried moving the computation inside the match, but the code had more right-ward drift than I thought was good, so kept it out for now